### PR TITLE
Move to more recent networkx dependency

### PR DESCRIPTION
**Pull request by sjsrey**

### CODE REVIEW

**Comment by renanxcortes** [comment]
Can https://github.com/pysal/region/pull/34 be closed in favor of this one?

File: setup.py [hunk 1]
```diff
@@ -5,44 +5,50 @@
...
+            "Programming Language :: Python :: 3",
+            "Programming Language :: Python :: 3.5",
+            "Programming Language :: Python :: 3.6",
+        ],
+        keywords="regionalization spatial clustering",
+        packages=find_packages(exclude=["contrib", "doc", "test*"]),
+        install_requires=[
+            "geopandas",
+            "libpysal",
+            "networkx>=2.0.0",
```
Comment (renanxcortes) [inline setup.py]: This is the actual only change in this commit, right?

Comment (sjsrey) [reply setup.py]: No. There are a bunch of changes. https://github.com/pysal/region/pull/35/files

Comment (renanxcortes) [reply setup.py]: But the others are related to code formatting, isn't it?

Comment (knaaptime) [reply setup.py]: i merged #34 for clarity in the diff. some of the changes are from black/reformatting but others look like they're for compatibility with the api from the newer networkx

Comment (sjsrey) [reply setup.py]: Most of them are black formatting, but there are substantive changes along the lines of https://github.com/pysal/region/pull/35/files#diff-0ebddef0485a8d8e6b6db081fcb2d99dR66

Comment (renanxcortes) [reply setup.py]: Ok, got it

**Pull request merged**